src/hooks/useSelector.ts

```diff
@@ -16,94 +19,29 @@ function useSelectorWithStoreAndSubscription<TStoreState, TSelectedState>(
   store: Store<TStoreState, AnyAction>,
   contextSub: Subscription
 ): TSelectedState {
-  const [, forceRender] = useReducer((s) => s + 1, 0)
-
-  const subscription = useMemo(
-    () => createSubscription(store, contextSub),
-    [store, contextSub]
-  )
-
-  const latestSubscriptionCallbackError = useRef<Error>()
-  const latestSelector = useRef<TSelector<TStoreState, TSelectedState>>()
-  const latestStoreState = useRef<TStoreState>()
-  const latestSelectedState = useRef<TSelectedState>()
+  const subscribe = useMemo(() => {
+    const subscription = createSubscription(store, contextSub)
```

Comment:
Since the plan is to require React 18 in the next version, can we remove this subscription indirection and subscribe directly to the store? Now that 18 has batching by default.

Reply:
The one other hypothetical piece that would potentially lose is that in theory you can avoid the "zombie child" problem by slapping a `connect()`-ed component above a `useSelector()`-ed component, in which case the selector subscribes to the connect's `Subscription`.

In practice I don't think anyone seriously does that.

Given that `Subscription` is like half of the total bytes here for `useSelector` anyway (like 1K), that also seems reasonable to do.

Reply:
I think we addressed the zombie child problem is fixed by the new strategy where we always re-run the selector in the render phase (if it changed), right?

That was the intent when we implemented useSyncExternalStore, at least. If you can think of an example where it breaks let me know so we can fix it upstream.

Reply:
Also a good point, and tbh I'm not sure if we actually _have_ tests for zombie child-related behavior in our codebase atm.  Worth trying to add something.